ports-of-call/robust_utils.hpp

```diff
@@ -64,13 +64,40 @@ PORTABLE_FORCEINLINE_FUNCTION auto ratio(const A &a, const B &b) {
   return a / (b + sgn(b) * SMALL<B>());
 }
 
+template <typename DataType>
+PORTABLE_FUNCTION constexpr DataType reldiff(const DataType actual,
+                                             const DataType expected) {
+  if (actual == expected) {
+    return 0;
+  } else if (expected == 0) {
+    // expected 0, got nonzero, define this to be 100% error
+    return 1;
```

Comment:
I think it's often useful to switch to an absolute error in this case, for example return `actual` here. But up to you.